openwisp_users/tasks.py

```diff
@@ -83,3 +85,20 @@ def password_expiration_email():
             sleep(random.randint(1, 2))
         else:
             email_counts += 1
+
+
+@shared_task
+def invalidate_org_membership_cache(organization_pk):
+    """
+    Invalidates organization membership cache of all users of an
+    organization when organization.is_active changes
+    (organization is disabled or enabled again).
+    """
+    qs = OrganizationUser.objects.filter(
+        organization_id=organization_pk
+    ).select_related('user')
+    User = get_user_model()
+    for user in qs.iterator():
+        if not isinstance(user, User):
+            user = user.user
+        user._invalidate_user_organizations_dict()
```

Comment:
Is all this really necessary? Why do you feel the need of checking if user.user is an instance of `User`? Could it be anything different? Let me know.

I think this could be simplified as following:

```suggestion
        user._invalidate_user_organizations_dict()
```

If I am missing something let me know.

Reply:
Yes the if condition can be removed but I cannot do this 
```
user._invalidate_user_organizations_dict()
```
This will cause error as organization user does not have this method. 
So we can do this 
```
user.user._invalidate_user_organizations_dict()
```

Reply:
Using the variable name `user` is misleading here, this is an OrganizationUser instance, so let's call it `org_user`.
Then you can call: `org_user.user._invalidate_user_organizations_dict()`.